apps/local_laplacian/local_laplacian_generator.cpp

```diff
@@ -124,17 +117,16 @@ class LocalLaplacian : public Halide::Generator<LocalLaplacian> {
             }
             outGPyramid[0]
                 .compute_at(output, y).vectorize(x, 8);
-            for (int j = 5; j < J; j++) {
+            for (int j = 5; j < pyramid_levels; j++) {
                 inGPyramid[j].compute_root();
                 gPyramid[j].compute_root().parallel(k);
                 outGPyramid[j].compute_root();
             }
         }
-
-        return output;
     }
 private:
     Var x, y, c, k;
+    Func remap, gray, inGPyramid[maxJ], gPyramid[maxJ], outGPyramid[maxJ];
```

Comment:
Yeah, the more I think about this the more I think it could solve a host of unpleasantries. I'll prototype something tomorrow and see how it feels in practice.